Wrappers/Python/ccpi/viewer/ui/main_windows.py

```diff
@@ -0,0 +1,782 @@
+
+    Parameters
+    ----------
+    title : str, optional
+        The title of the window. The default is "ViewerMainWindow".
+    app_name : str, optional
+        The name of the application. The default is None.
+    settings_name : str, optional
+        The name of the settings file. The default is None.
+    organisation_name : str, optional
+        The name of the organisation. The default is None.
+    '''
+
+    def __init__(self,
+                 title="ViewerMainWindow",
+                 app_name=None,
+                 settings_name=None,
+                 organisation_name=None,
+                 *args,
+                 **kwargs):
+
+        super(ViewerMainWindow, self).__init__(title,
+                                               app_name,
+                                               settings_name=settings_name,
+                                               organisation_name=organisation_name)
+
+        self.default_downsampled_size = 512**3
+
+        self.viewers = []
+        self.viewer_docks = []
+
+        self.createViewerCoordsDockWidget()
+
+    def createAppSettingsDialog(self):
+        '''Create a dialog to change the application settings.
+        This is a method in the MainWindowWithSessionManagement class, which we
+        override here to make our own settings dialog'''
+        dialog = ViewerSettingsDialog(self)
+        dialog.Ok.clicked.connect(lambda: self.onAppSettingsDialogAccepted(dialog))
+        dialog.Cancel.clicked.connect(dialog.close)
+        self.setAppSettingsDialogWidgets(dialog)
+        dialog.open()
+
+    def setAppSettingsDialogWidgets(self, dialog):
+        '''Set the widgets on the app settings dialog, based on the 
+        current settings of the app.
+        
+        Parameters
+        ----------
+        dialog : ViewerSettingsDialog
+            The dialog to set the widgets on.
+        '''
+        super().setAppSettingsDialogWidgets(dialog)
+        sw = dialog.widgets
+
+        if self.settings.value("vis_size") is not None:
+            sw['vis_size_field'].setValue(float(self.settings.value("vis_size")))
+        else:
+            sw['vis_size_field'].setValue(self.getDefaultDownsampledSize() / (1024**3))
+
+        if self.settings.value("use_gpu_volume_mapper") is not None:
+            use_gpu = str(self.settings.value("use_gpu_volume_mapper"))
+            sw['gpu_checkbox_field'].setChecked(use_gpu == "true")
+        else:
+            sw['gpu_checkbox_field'].setChecked(True)
+
+    def onAppSettingsDialogAccepted(self, settings_dialog):
+        '''This is called when the user clicks the OK button on the
+        app settings dialog. We override this method to save the
+        settings to the QSettings object.
+        
+        Parameters
+        ----------
+        settings_dialog : ViewerSettingsDialog
+            The dialog to get the settings from.
+        '''
+        super().onAppSettingsDialogAccepted(settings_dialog)
```

Comment:
I don't think there is a way to avoid calling super() here, because this is a method that is connected to the OK button of the dialog.

Reply:
resolved by adding multiple connections to the OK button